src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,304 @@
+            tasks.add(newDeadline);
+        } catch (Exception e) {
+            throw new DataFileCorruptedException();
+        }
+    }
+
+    public static void handleE (ArrayList<Task> tasks, String[] parameters) throws DataFileCorruptedException{
+        try {
+            String description = parameters[2];
+            String at = parameters[3];
+            Deadline newEvent = new Deadline(description, at);
+            newEvent.setStatus(parameters[1].equals("✓"));
+            tasks.add(newEvent);
+        } catch (Exception e) {
+            throw new DataFileCorruptedException();
+        }
+    }
+    public static void addToTasks(String input, ArrayList<Task> tasks) {
+        String[] parameters = input.split(" # ");
+        try {
+            switch (parameters[0]) {
+            case "T": {
+                handleT(tasks, parameters);
+                break;
+            }
+            case "D": {
+                handleD(tasks, parameters);
+                break;
+            }
+            case "E": {
+                handleE(tasks, parameters);
+                break;
+            }
+            default: {
+                System.out.println("Data file data/amon_gus.txt corrupted!");
+                break;
+            }
+            }
+        } catch (DataFileCorruptedException e) {
+            System.out.println("Data file data/amon_gus.txt corrupted!");
+        }
+
+    }
+
+
+    public static void readFile(ArrayList<Task> tasks) throws FileNotFoundException{
+        File f = new File(AMON_GUS_PATH);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            addToTasks(s.nextLine(), tasks);
+        }
+    }
+
+    public static void fileIO(ArrayList<Task> tasks) {
+        try {
+            readFile(tasks);
+        } catch (FileNotFoundException e) {
+            File f = new File (AMON_GUS_PATH);
+        }
+    }
+
+    public static void readInput() {
+        Scanner in = new Scanner(System.in);
+        String line = "";
+        ArrayList<Task> tasks = new ArrayList<>();
+        fileIO(tasks);
+        while (!line.equals("bye")) {
+            System.out.println(LINE_SEPARATOR);
+            System.out.println("To view my commands, type \"help\"\nEnter a Todo, Deadline, or Event: ");
+            line = in.nextLine();
+            handleCommand(line, tasks);
+        }
+    }
+
+    public static void run() {
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("I keep track of your crewmate tasks and find imposters! \nHow can i help you?");
+        readInput();
+        System.out.println(Helper.getRandomIntegerBetweenRange(0, 6) == 1 ?
+                "You're the imposter? （°0°） How could you!" :
+                "Yay we found the imposter! ヽ(^o^)ノ The ship is saved!");
+    }
+
+
+    public static void main(String[] args) {
+        System.out.println("Hello! I'm Amon Gus");
+        System.out.println(ASCIIconstants.logo);
+        run();
+    }
+}
```

Comment:
You have packed too many methods inside this one class. Please consider placing some of the methods in other classes. For example, methods for managing tasks can be under a class named "TaskManager".